### PR TITLE
Clean up CI and drop py3.7

### LINKED ISSUE #232: Broken with Sphinx 7.2
Sphinx 7.2 removed `path` from `sphinx.testing.util`.

https://github.com/sphinx-doc/sphinx/pull/11526 looks like you were supposed to use sphinx.testing.path, which is deprecated and will be removed in Sphinx 9

```
[   99s] ============================= test session starts ==============================
[   99s] platform linux -- Python 3.9.18, pytest-7.4.0, pluggy-1.0.0 -- /usr/bin/python3.9
[   99s] cachedir: .pytest_cache
[   99s] rootdir: /home/abuild/rpmbuild/BUILD/jupyter-sphinx-0.4.0
[  100s] collecting ... collected 0 items / 1 error
[  100s] 
[  100s] ==================================== ERRORS ====================================
[  100s] ____________________ ERROR collecting tests/test_execute.py ____________________
[  100s] ImportError while importing test module '/home/abuild/rpmbuild/BUILD/jupyter-sphinx-0.4.0/tests/test_execute.py'.
[  100s] Hint: make sure your test modules/packages have valid Python names.
[  100s] Traceback:
[  100s] /usr/lib64/python3.9/importlib/__init__.py:127: in import_module
[  100s]     return _bootstrap._gcd_import(name[level:], package, level)
[  100s] tests/test_execute.py:15: in <module>
[  100s]     from sphinx.testing.util import SphinxTestApp, assert_node, path
[  100s] E   ImportError: cannot import name 'path' from 'sphinx.testing.util' (/usr/lib/python3.9/site-packages/sphinx/testing/util.py)
[  100s] =========================== short test summary info ============================
[  100s] ERROR tests/test_execute.py
[  100s] !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
[  100s] =============================== 1 error in 0.42s ===============================
```

### PR DESCRIPTION
**Pull request by blink1073**
Python 3.7 is EOL.  Clean up CI and test all current Python versions.
Update readthedocs config.

Fixes https://github.com/jupyter/jupyter-sphinx/issues/230.  Fixes #232

### CODE REVIEW

**Comment by blink1073** [comment]
@akhmerov can you please add `blink1073` to https://readthedocs.org/projects/jupyter-sphinx/ so I can set up the PR build integration?

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,25 +1,36 @@
...
+      with:
+        python-version: 3.8
+    - uses: pre-commit/action@v2.0.0
+
   tests:
 
     strategy:
       matrix:
-        python-version: [3.7, 3.8]
+        python-version: ["3.8", "3.11", " 3.12"]
```
Comment (12rambau) [inline .github/workflows/tests.yml:20]: any reason why you are not testing all python version on ubuntu and only latest on extra platform (mac and windows). This is what I have been doing so far in other repo but now I'm questioning myself....

**Comment by 12rambau** [comment]
Fix #230

**Comment by 12rambau** [comment]
@akhmerov if you can add `12rambau` as well that would be good. AS it's a sphinx extension sometime you just want to retrigger the PR doc build instead of the whole test suit and you can only do that with admin rights.

**Comment by akhmerov** [comment]
@blink1073 done!

@12rambau I see this:

image

**Comment by akhmerov** [comment]
@blink1073 @12rambau thank you for stepping in!

**Comment by 12rambau** [comment]
> @blink1073 done!
> 
> @12rambau I see this:
> 
> image

You are right, I'm "borntobealive" over there (same knight avatar if it helps)

**Comment by akhmerov** [comment]
@blink1073 another RTD-related issue: the webhook integration [listed here](https://readthedocs.org/dashboard/jupyter-sphinx/integrations/) needs to be updated with a secret, and I cannot do this because of not having sufficient github permissions for this repo.

**Comment by blink1073** [comment]
Thanks @akhmerov, I created a new webhook, and it is working now

**Comment by 12rambau** [comment]
@blink1073, Is this ready to merge or you have other requests related to the build ?

**Comment by blink1073** [comment]
> you have other requests related to the build ?

Yes, but in follow up PRs.  :)

**Pull request merged**